packages/design-system/src/components/Table/Table.tsx

```diff
@@ -65,11 +47,6 @@ interface BaseTableProps {
    * A striped variation of the table.
    */
   striped?: boolean;
-  /**
-   * Disables the warning message on development console when a responsive stackable table cell does not contain an `id` or `headers`.
-   * It's recommended that accessibility with screen readers is tested to ensure the stacked table meets the requirement.
-   */
-  warningDisabled?: boolean;
```

Comment:
I think I have a TS question here... TableCell has optional props (`header` and `id`) that are actually required, but only if another prop is either a `td` or a `th`. I've heard there a way to create a specific type that fits this circumstance (I think it's called a [conditional type](https://www.typescriptlang.org/docs/handbook/2/conditional-types.html)?) and I wonder if it's a better approach than having these warning props?

Something like:

```js
interface BaseTableCellProps {
  component?: 'th' | 'td';
}

interface TDProps {
  header: string;
}

interface THProps {
  id: string;
}

type TableCellProps = BaseTableCellProps extends 'th' ? THProps : TDProps;
```

I'm not sure if this is a better implementation or not, so discussion would be appreciated. It just seems we have a giant bucket of props that only apply if certain other props are set and it might be cleaner to split them up?